sidestep_engine/core/trainer_helpers.py

```
@@ -692,7 +691,37 @@ def resume_checkpoint(
+                from peft import set_peft_model_state_dict
+                set_peft_model_state_dict(decoder, state_dict)
+                loaded_via_peft = True
+            except Exception as exc:
+                logger.debug("PEFT set_peft_model_state_dict failed (%s), trying manual remap", exc)
+
+            if not loaded_via_peft:
+                # Fallback: manually re-insert '.default.' into LoRA keys
+                remapped = {}
+                for k, v in state_dict.items():
+                    if ".lora_A." in k and ".lora_A.default." not in k:
+                        k = k.replace(".lora_A.", ".lora_A.default.")
```

Ruff (0.15.7)

[warning] `for` loop variable `k` overwritten by assignment target

(PLW2901)